nautobot/dcim/tests/test_models.py

```diff
@@ -319,13 +266,9 @@ def test_change_rackgroup_location_children_not_permitted(self):
         location_c = Location.objects.create(name="Location C", location_type=location_type_c, parent=self.location_a)
 
         self.rackgroup_a1.location = location_c
-        self.rackgroup_a1.save()
-
-        self.assertEqual(RackGroup.objects.get(pk=self.rackgroup_a1.pk).location, location_c)
-        self.assertEqual(RackGroup.objects.get(pk=self.rackgroup_a2.pk).location, location_c)
-        self.assertEqual(Rack.objects.get(pk=self.rack1.pk).location, None)
-        self.assertEqual(Rack.objects.get(pk=self.rack2.pk).location, None)
-        self.assertEqual(PowerPanel.objects.get(pk=self.powerpanel1.pk).location, None)
+        with self.assertRaises(ValidationError) as cm:
+            self.rackgroup_a1.save()
+        self.assertIn(f'Racks may not associate to locations of type "{location_type_c}"', str(cm.exception))
```

Comment:
Good to assert that none of the RackGroup/Rack/PowerPanel objects got updated to the new location?